tests/network/bindingplugin_passt.go

```diff
@@ -331,5 +336,86 @@ EOL`, inetSuffix, serverIP, serverPort)
+					libvmi.WithInterface(libvmi.InterfaceWithPasstBindingPlugin()),
+					libvmi.WithNetwork(v1.DefaultPodNetwork()),
+					libvmi.WithCloudInitNoCloudNetworkData(networkData),
+				)
+				vmi, err = kubevirt.Client().VirtualMachineInstance(testsuite.GetTestNamespace(nil)).Create(context.Background(), vmi)
+				ExpectWithOffset(1, err).ToNot(HaveOccurred())
+				vmi = libwait.WaitUntilVMIReady(vmi,
+					console.LoginToFedora,
+					libwait.WithFailOnWarnings(false),
+					libwait.WithTimeout(180),
+				)
+				return vmi
+			}
+
+			By("Starting a VMI")
+			migrateVMI := startPasstVMI(libvmi.NewFedora, console.LoginToFedora)
+			beforeMigNodeName := migrateVMI.Status.NodeName
+
+			By("Starting another VMI")
+			anotherVMI := startPasstVMI(libvmi.NewAlpine, console.LoginToAlpine)
+
+			By("Verify the VMIs can ping each other")
+			migrateVmiBeforeMigIP := libnet.GetVmiPrimaryIPByFamily(migrateVMI, ipFamily)
+			anotherVmiIP := libnet.GetVmiPrimaryIPByFamily(anotherVMI, ipFamily)
+			Expect(libnet.PingFromVMConsole(migrateVMI, anotherVmiIP)).To(Succeed())
+			Expect(libnet.PingFromVMConsole(anotherVMI, migrateVmiBeforeMigIP)).To(Succeed())
+
+			By("Perform migration")
+			migration := libmigration.New(migrateVMI.Name, migrateVMI.Namespace)
+			migrationUID := libmigration.RunMigrationAndExpectToCompleteWithDefaultTimeout(kubevirt.Client(), migration)
+			migrateVMI = libmigration.ConfirmVMIPostMigration(kubevirt.Client(), migrateVMI, migrationUID)
+
+			By("Verify all the containers in the source pod were terminated")
+			labelSelector := fmt.Sprintf("%s=%s", v1.CreatedByLabel, string(migrateVMI.GetUID()))
+			fieldSelector := fmt.Sprintf("spec.nodeName==%s", beforeMigNodeName)
+
+			Eventually(func() bool {
+				pods, err := kubevirt.Client().CoreV1().Pods(migrateVMI.Namespace).List(context.Background(),
+					metav1.ListOptions{LabelSelector: labelSelector, FieldSelector: fieldSelector},
+				)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(len(pods.Items)).To(Equal(1))
+
+				return pods.Items[0].Status.Phase == k8sv1.PodSucceeded
+			}, 30*time.Second).Should(BeTrue())
+
+			By("Verify the VMI new IP is propogated to the status")
+			var migrateVmiAfterMigIP string
+			Eventually(func() bool {
+				migrateVMI, err = kubevirt.Client().VirtualMachineInstance(migrateVMI.Namespace).Get(context.Background(), migrateVMI.Name, &metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred(), "should have been able to retrive the VMI instance")
+				migrateVmiAfterMigIP = libnet.GetVmiPrimaryIPByFamily(migrateVMI, ipFamily)
+				return migrateVmiBeforeMigIP != migrateVmiAfterMigIP
+			}, 30*time.Second).Should(BeTrue(), "the VMI status should get a new IP after migration")
+
+			By("Verify the guest renewed the IP")
+			ipValidation := fmt.Sprintf("ip addr show eth0 | grep %s", migrateVmiAfterMigIP)
+			Eventually(func() error {
+				return console.RunCommand(migrateVMI, ipValidation, time.Second*5)
+			}, 30*time.Second).Should(Succeed())
```

Comment:
Why do we care?
It may have not renewed, just stayed the same. No?

If it is wrong, the ping later will fail and we will know the actual IP from the collected data.

Reply:
The ping won't fail, everything works perfectly even if the IP of the old pod remains.
We renew the IP just to avoid the discussed problem with keeping the old IP -
- If another pod in the cluster get the IP of the old pod.
- Some users want the in and out IP to be same.

Reply:
Ok, so can you please comment this here? I am sure I will forget about the reasoning in a week again.

Reply:
It is part of the design doc, I also will add it to the PR message. It doesn;t make sense to have it in the test as a comment.